### PR TITLE
MSEARCH-380 Browse contributors: listener not working for new tenants

### PR DESCRIPTION
**Pull request by viacheslavkol**
### Purpose
Fix contributors saving for new tenants.

### Approach
- remove kafka-topics.json
- add mod-search-specific topics to application.yml configuration
- rework topics for new tenant creation with data from application.yml
- update tests

### Learning
[MSEARCH-380](https://issues.folio.org/browse/MSEARCH-380)

### CODE REVIEW

File: src/main/java/org/folio/search/configuration/properties/FolioKafkaProperties.java [hunk 1]
```diff
@@ -49,4 +57,25 @@ public static class KafkaListenerProperties {
...
+  @AllArgsConstructor(staticName = "of")
+  public static class KafkaTopic {
+
+    /**
+     * Topic name.
+     */
+    private String name;
+
+    /**
+     * Number of partitions for topic as spring placeholder expression.
```
Comment (psmagin) [inline src/main/java/org/folio/search/configuration/properties/FolioKafkaProperties.java:72]:
```suggestion
     * Number of partitions for topic.
```

File: src/main/java/org/folio/search/configuration/properties/FolioKafkaProperties.java [hunk 2]
```diff
@@ -49,4 +57,25 @@ public static class KafkaListenerProperties {
...
+     */
+    private Integer numPartitions;
+
+    /**
+     * Replication factor for topic as spring placeholder expression.
```
Comment (psmagin) [inline src/main/java/org/folio/search/configuration/properties/FolioKafkaProperties.java:77]:
```suggestion
     * Replication factor for topic 
```

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -75,6 +75,10 @@ application:
   kafka:
     retry-interval-ms: ${KAFKA_RETRY_INTERVAL_MS:2000}
     retry-delivery-attempts: ${KAFKA_RETRY_DELIVERY_ATTEMPTS:6}
+    topics:
+      - name: search.instance-contributor
+        numPartitions: ${KAFKA_EVENT_TOPICS_PARTITIONS:50}
+        replicationFactor: ${KAFKA_EVENT_TOPICS_REPLICATION_FACTOR:}
```
Comment (psmagin) [inline src/main/resources/application.yml:81]:
```suggestion
        numPartitions: ${KAFKA_CONTIBUTORS_TOPIC_PARTITIONS:50}
        replicationFactor: ${KAFKA_CONTIBUTORS_TOPIC_REPLICATION_FACTOR:}
```
You change it in documentation.

File: src/main/resources/kafka/kafka-topics.json [hunk 4]
```diff
@@ -1,5 +0,0 @@
-{
-  "topics": [
-
-  ]
-}
```
Comment (psmagin) [inline src/main/resources/kafka/kafka-topics.json:5]: Any mentions of the file in the documentation for the module?

File: README.md [hunk 5]
```diff
@@ -151,41 +151,43 @@ with less powerful configuration (see [High availability](https://www.elastic.co
...
+| KAFKA_AUTHORITIES_CONSUMER_PATTERN          | -                         | Custom subscription pattern for Kafka authority message consumers.                                                                                                                    |
+| KAFKA_AUTHORITIES_CONCURRENCY               | 1                         | Custom number of kafka concurrent threads for authority message consuming.                                                                                                            |                                                                                                            |
+| KAFKA_CONTRIBUTORS_TOPIC_PARTITIONS         | 50                        | Amount of partitions for contributors topic.                                                                                                                                          |
+| KAFKA_CONTRIBUTORS_TOPIC_REPLICATION_FACTOR | -                         | Replication factor for contributors topic.                                                                                                                                            |
```
Comment (psmagin) [inline README.md:178]:
```suggestion
| KAFKA_CONTRIBUTORS_TOPIC_PARTITIONS         | 50                        | Amount of partitions for `search.instance-contributor` topic.                                                                                                                                          |
| KAFKA_CONTRIBUTORS_TOPIC_REPLICATION_FACTOR | -                         | Replication factor for `search.instance-contributor` topic.                                                                                                                                            |
```

File: src/main/resources/application.yml (hunk 3 above)
Comment (psmagin) [inline src/main/resources/application.yml:81]: Let's describe these properties somewhere in the documentation.

**Review by psmagin — CHANGES_REQUESTED** [review]

File: src/main/resources/kafka/kafka-topics.json (hunk 4 above)
Comment (viacheslavkol) [reply src/main/resources/kafka/kafka-topics.json:5]: No mentions in any of project files (except in KafkaAdminService's comment that should be removed).

**Review by psmagin — APPROVED** [review]

**Review by Shans-Kaluhin — APPROVED** [review]

**Pull request merged**